homeassistant/components/system_log.py

```diff
@@ -0,0 +1,127 @@
+"""
+Support for system log.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/system_log/
+"""
+import os
+import re
+import asyncio
+import traceback
+from io import StringIO
+
+import voluptuous as vol
+
+from homeassistant.bootstrap import DATA_SYSTEM_LOG
+from homeassistant.config import load_yaml_config_file
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.http import HomeAssistantView
+from homeassistant.util.logging import LogErrorHandler
+
+DOMAIN = 'system_log'
+DEPENDENCIES = ['http']
+SERVICE_CLEAR = 'clear'
+
+CONF_MAX_ENTRIES = 'max_entries'
+
+DEFAULT_MAX_ENTRIES = 50
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Optional(CONF_MAX_ENTRIES,
+                     default=DEFAULT_MAX_ENTRIES): cv.positive_int,
+    }),
+}, extra=vol.ALLOW_EXTRA)
+
+SERVICE_CLEAR_SCHEMA = vol.Schema({})
+
+
+@asyncio.coroutine
+def async_setup(hass, config):
+    """Set up the logger component."""
+    conf = config.get(DOMAIN)
+
+    if conf is None:
+        conf = CONFIG_SCHEMA({})
+
+    if DATA_SYSTEM_LOG not in hass.data:
```

Comment:
We should fail setup if this is not set. Setting it in hass.data without being connected to the logger makes no sense.

Actually, why do we even set something like this in bootstrap? I'm thinking now that all code related to setting up a logger etc can live in this component. No need to touch the core.

Reply:
I followed the advice from @pvizeli to add it to core and I think that's the easiest way to catch everything. It would of course be possible to create a new handler and add that to the root logger from here, but that would mean that we might miss log points that happens before this component is setup. Some of those errors are of course fatal and will not make Home Assistant start. In that case it's fine, but we might have other issues that we miss. The warning about python 3.4 being deprecated is one such example, that was missing before I moved the log handler to bootstrap. Do you have any further suggestions?

Reply:
Oh, and some tests failed because of `DATA_SYSTEM_LOGGER` not being available in `hass.data`. So I guess bootstrap was not run in those tests cases. I can remove it and see what happens.

Reply:
We have a priority system for components. Would that work: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/bootstrap.py#L32-L34

Yeah, you might miss maybe 1 or 2 logs but we also keep all the code consolidated.